src/abstractitensornetwork.jl

```diff
@@ -879,14 +881,18 @@ end
 function add(tn1::AbstractITensorNetwork, tn2::AbstractITensorNetwork)
   @assert issetequal(vertices(tn1), vertices(tn2))
 
-  #Check for any edges with multiple indices and combine to avoid
-  if has_multi_edge(tn1)
-    tn1 = combine_linkinds(tn1)
-  end
-
-  if has_multi_edge(tn2)
-    tn2 = combine_linkinds(tn2)
-  end
+  tn1 = combine_linkinds(
+    tn1,
+    linkinds_combiners(
+      tn1; edges_to_combine=filter(e -> length(linkinds(tn1, e)) > 1, edges(tn1))
+    ),
+  )
```

Comment:
Seems simpler to just add an `edges` keyword argument directly to `combine_linkinds`.

Reply:
Mmmm but `combine_linkinds` by default populates a `combiner` along every edge of the network and I want to avoid it doing that? So like I originally had
`combine_linkinds(tn::AbstractITensorNetwork, combiners=linkinds_combiners(tn); edges = edges(tn)` but then I don't want the user to call `combine_linkinds(tn; edges = edge_subset)` which would create combiners on every edge and end up not using them, making them redundant. Unless it's possible to do
`combine_linkinds(tn, combiners=linkinds_combiners(tn; edges); edges = edges(tn)`?